packages/codemods/src/codeMods/mods/PersonaToAvatar/PersonaToAvatar.mod.ts

```diff
@@ -0,0 +1,179 @@
+                    });
+                    break;
+                  }
+                }
+              }
+            }
+          }
+        }
+      });
+    }
+  });
+}
+
+export function replacePersonaImport(file: SourceFile) {
+  let found = false;
+  file.getImportDeclarations().forEach(imp => {
+    if (imp.getModuleSpecifierValue() === personaPath) {
+      imp.getNamedImports().forEach(val => {
+        if (val.getText() === 'Persona') {
+          found = true;
+          val.renameAlias('Avatar');
+          val.remove();
+        }
+      });
+    }
+  });
+  if (found) {
+    appendOrCreateNamedImport(file, 'office-ui-fabric-react/lib/Avatar', ['Avatar']);
+  }
+}
+
+export function replaceIPersonaPropsImport(file: SourceFile) {
+  // Figure out if I should actually make this change
+  // TODO need to test with a variety of things, maybe one that serves as a passthrough
+  let found = false;
+  file.getImportDeclarations().forEach(imp => {
+    if (imp.getModuleSpecifierValue() === personaPath) {
+      imp.getNamedImports().forEach(val => {
+        if (val.getText() === 'IPersonaProps') {
+          val.renameAlias('AvatarProps');
+          val.remove();
+          found = true;
+        }
+      });
+    }
+  });
+  if (found) {
+    appendOrCreateNamedImport(file, 'office-ui-fabric-react/lib/Avatar', ['AvatarProps']);
+  }
+}
+
+export function replacePersonaSizeImport(file: SourceFile) {
+  let found = false;
+  file.getImportDeclarations().forEach(imp => {
+    if (imp.getModuleSpecifierValue() === personaPath) {
+      imp.getNamedImports().forEach(val => {
+        if (val.getText() === 'PersonaSize') {
+          found = true;
+          val.renameAlias('AvatarSize');
+          val.remove();
+        }
+      });
+    }
+  });
+  if (found) {
+    appendOrCreateNamedImport(file, 'office-ui-fabric-react/lib/Avatar', ['AvatarSize']);
+  }
+}
+
+// Gets the parent that is a direct descendant of a block
+// Which should allow for better inserting
+function getBlockContainer(node: Node<ts.Node>) {
+  return node.getFirstAncestor(ans => {
+    const ansPar = ans.getParent();
+    return ansPar?.getKind() === SyntaxKind.Block;
+  });
+}
+
+export function renamePrimaryTextProp(file: SourceFile) {
+  // Should this fix the naming if the Persona Component has already been renamed to Avatar
+  const elements = findJsxTag(file, 'Persona');
+  renameProperty(elements, 'primaryText', 'text');
+}
+
+export function renameRenderCoin(file: SourceFile) {
+  // Should this fix the naming if the Persona Component has already been renamed to Avatar
+
+  const elements = findJsxTag(file, 'Persona');
+  renameProperty(elements, 'onRenderCoin', 'onRenderAvatarCoin');
+}
+
+const PersonaToAvatarMod: ICodeMod = {
+  run: (file: SourceFile) => {
+    try {
+      replacePersonaImport(file);
+      replaceIPersonaPropsImport(file);
+      replacePersonaSizeImport(file);
+      renamePrimaryTextProp(file);
+      renameRenderCoin(file);
```

Comment:
These right here; in my suggestion above, could these be just replacement types in a config?

Some of them probably not. E.g. `renameRenderCoin` finds `Persona` jsx elements and renames `onRenderCoin` to `onRenderAvatarCoin` (which I think is wrong btw.)

Reply:
So, you may need a custom `type` which does some sort of manual work programatically. But whatever we can do to simplify and keep the codemod data driven is good.

Reply:
This is more of a proof of concept for right now that I should make more clear. 

As for the config, we could potentially allow some of these to take in a json blob. For the start we should get the api of the functions correct. Trip and I are working on that.